core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -2904,15 +2911,15 @@ struct ViewValueFunctor<ExecSpace, ValueType, true /* is_scalar */> {
   ViewValueFunctor& operator=(const ViewValueFunctor&) = default;
 
   ViewValueFunctor(ExecSpace const& arg_space, ValueType* const arg_ptr,
-                   size_t const arg_n)
-      : space(arg_space), ptr(arg_ptr), n(arg_n) {}
+                   size_t const arg_n, std::string arg_name)
+      : space(arg_space), ptr(arg_ptr), n(arg_n), name(std::move(arg_name)) {}
 
   void construct_shared_allocation() {
     if (!space.in_parallel()) {
-      uint64_t kpID = 0;
+      std::string functor_name = "Kokkos::View::initialization [" + name + "]";
```

Comment:
Same comment as above about scoping, only this time I don't see the reason to even declare the variable.